src/analysis/retail/mage/fire/guide/Combustion.tsx

```diff
@@ -0,0 +1,234 @@
+import { ReactNode } from 'react';
+import SPELLS from 'common/SPELLS';
+import TALENTS from 'common/TALENTS/mage';
+import { SpellIcon, SpellLink, TooltipElement } from 'interface';
+import Analyzer from 'parser/core/Analyzer';
+import { RoundedPanel } from 'interface/guide/components/GuideDivs';
+import { BoxRowEntry, PerformanceBoxRow } from 'interface/guide/components/PerformanceBoxRow';
+import { explanationAndDataSubsection } from 'interface/guide/components/ExplanationRow';
+import { QualitativePerformance } from 'parser/ui/QualitativePerformance';
+import { qualitativePerformanceToColor } from 'interface/guide';
+import { PerformanceMark } from 'interface/guide';
+import CastEfficiencyBar from 'parser/ui/CastEfficiencyBar';
+import { GapHighlight } from 'parser/ui/CooldownBar';
+import { GUIDE_CORE_EXPLANATION_PERCENT } from 'analysis/retail/mage/fire/Guide';
+import CombustionActiveTime from '../core/CombustionActiveTime';
+import CombustionCasts from '../core/Combustion';
+import FeelTheBurn from '../talents/FeelTheBurn';
+
+class CombustionGuide extends Analyzer {
+  static dependencies = {
+    combustionActiveTime: CombustionActiveTime,
+    combustion: CombustionCasts,
+    feelTheBurn: FeelTheBurn,
+  };
+
+  protected combustionActiveTime!: CombustionActiveTime;
+  protected combustion!: CombustionCasts;
+  protected feelTheBurn!: FeelTheBurn;
+
+  generateGuideTooltip(
+    performance: QualitativePerformance,
+    tooltipText: ReactNode,
+    timestamp: number,
+  ) {
+    const tooltip = (
+      <>
+        <div>
+          <b>@ {this.owner.formatTimestamp(timestamp)}</b>
+        </div>
+        <div>
+          <PerformanceMark perf={performance} /> {performance}: {tooltipText}
+        </div>
+      </>
+    );
+    return tooltip;
+  }
+
+  get castDelay() {
+    const castDelay = this.combustion.combustionCastDelayThresholds.actual;
+    const minorThreshold = this.combustion.combustionCastDelayThresholds.isGreaterThan.minor;
+    const averageThreshold = this.combustion.combustionCastDelayThresholds.isGreaterThan.average;
+    const majorThreshold = this.combustion.combustionCastDelayThresholds.isGreaterThan.major;
+    let performance = QualitativePerformance.Fail;
+    if (castDelay < minorThreshold) {
+      performance = QualitativePerformance.Perfect;
+    } else if (castDelay < averageThreshold) {
+      performance = QualitativePerformance.Good;
+    } else if (castDelay < majorThreshold) {
+      performance = QualitativePerformance.Ok;
+    }
+    return performance;
+  }
```

Comment:
Seems like unnecessary assignments, perhaps just shorthand it by something like this would be cleaner.

```suggestion
  get castDelay() {
    const thresholds = this.combustion.combustionCastDelayThresholds;
    let performance = QualitativePerformance.Fail;
    if (castDelay < thresholds.minor) {
      performance = QualitativePerformance.Perfect;
    } else if (castDelay < thresholds.average) {
      performance = QualitativePerformance.Good;
    } else if (castDelay < thresholds.major) {
      performance = QualitativePerformance.Ok;
    }
    return performance;
  }

Reply:
Yea I like that better. I'll adjust it